learning_observer/learning_observer/auth/social_sso.py

```diff
@@ -143,6 +169,20 @@ async def social_handler(request):
     return aiohttp.web.HTTPFound(url)
 
 
+async def handle_lms_request_authorization(request):
```

Comment:
I'm not the biggest fan of the function name. Initially, I had `determine_roster_source` in mind. For now, this would probably be just a single line function returning `roster_source` from pmss. Eventually this would extract some information about the user and include that when querying the pmss settings.
We could keep the handling portions as a separate function (perhaps `_set_lms_header_information` or similar) once we've determined the roster source. This should probably be a dictionary mapping roster sources to handler functions (i.e. `_canvas`).

With `'canvas'` being used in quite a few places, it would be worth making it a constant.

Reply:
Fixed and also made 'canvas' a constant